### PR TITLE
Create project Files.Filesystem

### PR DESCRIPTION
**Pull request by cinqmilleans**
This pr add the new project Files.Filesystem and some helpers. This pr is part of a larger refactor, cut out for ease.

- Added project Files.Filesystem
- Move ILocalizationService to Shared (necessary).
- Move CommonPath and PathNormalization and FtpHelpers and EnumConversionHelpers.
- Helpers are improved (style, name, extension, ...). This simplifies calls.

**Validation**
How did you test these changes?
- [ ] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

File: src/Files.Filesystem/Files.Filesystem.csproj [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    <OutputPath>bin\x86\Debug\</OutputPath>
+    <DefineConstants>DEBUG;TRACE;NETFX_CORE;WINDOWS_UWP</DefineConstants>
+    <NoWarn>;2008</NoWarn>
+    <DebugType>full</DebugType>
+    <UseVSHostingProcess>false</UseVSHostingProcess>
+    <ErrorReport>prompt</ErrorReport>
+    <LangVersion>9</LangVersion>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x86'">
```
Comment (yaichenbaum) [inline src/Files.Filesystem/Files.Filesystem.csproj:57]: Can you add sideload configurations (should be similar to release)?

Comment (cinqmilleans) [reply src/Files.Filesystem/Files.Filesystem.csproj:68]: It's done.

File: src/Files.Backend/Models/ByteSize.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using Files.Backend.Extensions;
```
Comment (Fl436341) [inline src/Files.Backend/Models/ByteSize.cs:1]: src/Files.Backend/Models/ByteSize.cs

File: src/Files.Filesystem/Helpers/ConflitExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿using Files.Shared.Enums;
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/ConflitExtensions.cs:1]: src/Files.Filesystem/Helpers/ConflitExtensions.cs

File: src/Files.Filesystem/Helpers/ConflitExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+using Windows.Storage;
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/ConflitExtensions.cs:2]: src/Files.Filesystem/Helpers/ConflitExtensions.cs

File: src/Files.Filesystem/Helpers/ConflitExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Files.Filesystem.Helpers
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/ConflitExtensions.cs:4]: src/Files.Filesystem/Helpers/ConflitExtensions.cs

File: src/Files.Filesystem/Helpers/ConflitExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public static class ConflitExtensions
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/ConflitExtensions.cs:6]: src/Files.Filesystem/Helpers/ConflitExtensions.cs

File: src/Files.Filesystem/Helpers/ConflitExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        public static NameCollisionOption Convert(this FileNameConflictResolveOptionType option) => option switch
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/ConflitExtensions.cs:8]: src/Files.Filesystem/Helpers/ConflitExtensions.cs

File: src/Files.Filesystem/Helpers/ConflitExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            FileNameConflictResolveOptionType.Skip => NameCollisionOption.FailIfExists,
+            FileNameConflictResolveOptionType.GenerateNewName => NameCollisionOption.GenerateUniqueName,
+            FileNameConflictResolveOptionType.ReplaceExisting => NameCollisionOption.ReplaceExisting,
+            _ => NameCollisionOption.GenerateUniqueName,
+        };
+        public static FileNameConflictResolveOptionType ConvertBack(this NameCollisionOption option) => option switch
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/ConflitExtensions.cs:15]: src/Files.Filesystem/Helpers/ConflitExtensions.cs

File: src/Files.Filesystem/Helpers/ConflitExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            NameCollisionOption.FailIfExists => FileNameConflictResolveOptionType.Skip,
+            NameCollisionOption.GenerateUniqueName => FileNameConflictResolveOptionType.GenerateNewName,
+            NameCollisionOption.ReplaceExisting => FileNameConflictResolveOptionType.ReplaceExisting,
+            _ => FileNameConflictResolveOptionType.GenerateNewName,
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/ConflitExtensions.cs:20]: src/Files.Filesystem/Helpers/ConflitExtensions.cs

File: src/Files.Filesystem/Helpers/Constants.cs [hunk 10]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Files.Filesystem.Helpers
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/Constants.cs:1]: src/Files.Filesystem/Helpers/Constants.cs

File: src/Files.Filesystem/Helpers/Constants.cs [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+{
+    public static class Constants
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/Constants.cs:3]:
```suggestion
    public static class Constants
```

Comment (Fl436341) [reply src/Files.Filesystem/Helpers/Constants.cs:3]:
```suggestion
    public static class Constants
```src/Files.Filesystem/Helpers/Constants.cs

Comment (Fl436341) [inline src/Files.Filesystem/Helpers/Constants.cs:3]:
```suggestion
    public static class Constants
```

Comment (Fl436341) [reply src/Files.Filesystem/Helpers/Constants.cs:3]: src/Files.Filesystem/Helpers/Constants.cs

Comment (Fl436341) [reply src/Files.Filesystem/Helpers/Constants.cs:3]: src/Files.Filesystem/Helpers/Constants.cs src/Files.

File: src/Files.Filesystem/Helpers/FtpExtensions.cs [hunk 12]
```diff
@@ -0,0 +1,92 @@
+﻿using FluentFTP;
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/FtpExtensions.cs:1]: src/Files.Filesystem/Helpers/FtpExtensions.cs

File: src/Files.Filesystem/Helpers/FtpExtensions.cs [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+using System;
+using System.Threading.Tasks;
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/FtpExtensions.cs:3]: src/Files.Filesystem/Helpers/FtpExtensions.cs

File: src/Files.Filesystem/Helpers/Constants.cs (hunk 11 above)
Comment (Fl436341) [reply src/Files.Filesystem/Helpers/Constants.cs:3]: src/Files.Filesystem/Helpers/Constants.cs

File: src/Files.Uwp/Files.Uwp.csproj [hunk 14]
```diff
@@ -1695,4 +1695,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
+</Project>
```
Comment (Fl436341) [inline src/Files.Uwp/Files.Uwp.csproj:1698]: src/Files.Uwp/Files.Uwp.csproj

File: src/Files.Uwp/Files.Uwp.csproj [hunk 15]
```diff
@@ -1695,4 +1695,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
```
Comment (Fl436341) [inline src/Files.Uwp/Files.Uwp.csproj:1698]: src/Files.Uwp/Files.Uwp.csproj

File: src/Files.Shared/Files.Shared.csproj [hunk 16]
```diff
@@ -10,6 +10,7 @@
   </PropertyGroup>
 
   <ItemGroup>
```
Comment (Fl436341) [inline src/Files.Shared/Files.Shared.csproj:12]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Shared/Files.Shared.csproj:12]: src/Files.Shared/Files.Shared.csproj

File: src/Files.Shared/Files.Shared.csproj [hunk 17]
```diff
@@ -10,6 +10,7 @@
...
+    <PackageReference Include="CommunityToolkit.Mvvm" Version="8.0.0-preview4" />
```
Comment (Fl436341) [inline src/Files.Shared/Files.Shared.csproj:13]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Shared/Files.Shared.csproj:13]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Shared/Files.Shared.csproj:13]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Shared/Files.Shared.csproj:13]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Shared/Files.Shared.csproj:13]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [inline src/Files.Shared/Files.Shared.csproj:13]: src/Files.Shared/Files.Shared.csproj

File: src/Files.Filesystem/Files.Filesystem.csproj [hunk 18]
```diff
@@ -0,0 +1,240 @@
...
+  </PropertyGroup>
+  <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.CSharp.targets" />
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+       Other similar extension points exist, see Microsoft.Common.targets.
+  <Target Name="BeforeBuild">
+  </Target>
+  <Target Name="AfterBuild">
+  </Target>
+  -->
+</Project>
```
Comment (Fl436341) [inline src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Filesystem/Files.Filesystem.csproj

Comment (Fl436341) [inline src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Filesystem/Files.Filesystem.csproj

Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Shared/Files.Shared.csproj

Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Filesystem/Files.Filesystem.csproj

Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Filesystem/Files.Filesystem.csproj

File: src/Files.Filesystem/Files.Filesystem.csproj [hunk 19]
```diff
@@ -0,0 +1,240 @@
...
+    <VisualStudioVersion>14.0</VisualStudioVersion>
+  </PropertyGroup>
+  <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.CSharp.targets" />
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+       Other similar extension points exist, see Microsoft.Common.targets.
+  <Target Name="BeforeBuild">
+  </Target>
+  <Target Name="AfterBuild">
+  </Target>
+  -->
```
[the remark this answers is not in the document]
Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:239]: src/Files.Filesystem/Files.Filesystem.csproj

[the remark this answers is not in the document]
Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:239]: src/Files.Filesystem/Files.Filesystem.csproj

[the remark this answers is not in the document]
Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:239]: src/Files.Filesystem/Files.Filesystem.csproj

[the remark this answers is not in the document]
Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:239]: src/Files.Backend/ViewModels/Dialogs/AddItemDialog/AddItemDialogViewModel.cs

File: src/Files.Filesystem/Files.Filesystem.csproj (hunk 18 above)
Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Filesystem/Files.Filesystem.csproj

File: src/Files.Filesystem/Files.Filesystem.csproj (hunk 19 above)
[the remark this answers is not in the document]
Comment (Fl436341) [reply src/Files.Filesystem/Files.Filesystem.csproj:239]: src/Files.Filesystem/Files.Filesystem.csproj

File: src/Files.Filesystem/Files.Filesystem.csproj (hunk 18 above)
Comment (Fl436341) [inline src/Files.Filesystem/Files.Filesystem.csproj:240]: src/Files.Filesystem/Files.Filesystem.csproj

File: src/Files.Filesystem/Helpers/CommonPaths.cs [hunk 20]
```diff
@@ -0,0 +1,33 @@
+﻿using Files.Shared.Extensions;
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/CommonPaths.cs:1]: src/Files.Filesystem/Helpers/CommonPaths.cs

Comment (Fl436341) [inline src/Files.Filesystem/Helpers/CommonPaths.cs:1]: src/Files.Filesystem/Helpers/CommonPaths.cs

File: src/Files.Filesystem/Helpers/CommonPaths.cs [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+using System;
```
Comment (Fl436341) [inline src/Files.Filesystem/Helpers/CommonPaths.cs:2]: src/Files.Filesystem/Helpers/CommonPaths.cs

Comment (Fl436341) [inline src/Files.Filesystem/Helpers/CommonPaths.cs:2]: src/Files.Filesystem/Helpers/CommonPaths.cs

**Review by Fl436341 — APPROVED** [review]
src/Files.Filesystem/Helpers/CommonPaths.cs

File: src/Files.Backend/Models/ByteSize.cs (hunk 2 above)
Comment (Fl436341) [reply src/Files.Backend/Models/ByteSize.cs:1]: https://github.com/files-community/Files/pull/9486#pullrequestreview-1078249723

Comment (Fl436341) [reply src/Files.Backend/Models/ByteSize.cs:1]: https://github.com/files-community/Files/pull/9486#pullrequestreview-1078249723

File: src/Files.Filesystem/Helpers/ConflitExtensions.cs (hunk 3 above)
Comment (Fl436341) [reply src/Files.Filesystem/Helpers/ConflitExtensions.cs:1]: https://github.com/files-community/Files/pull/9486#discussion_r949781655

**Review by onein528 — CHANGES_REQUESTED** [review]
On `Files.Uwp/Views/MainPage.xaml.cs` around L95

Can you change "GetLocalized" to "GetLocalizedResource" to fix automation build errors?

**Pull request closed without merging**